cmd/bnscli/cmd_key.go

```diff
@@ -0,0 +1,88 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+
+	"github.com/iov-one/weave/crypto"
+	"golang.org/x/crypto/ed25519"
+)
+
+func cmdKeygen(input io.Reader, output io.Writer, args []string) error {
+	fl := flag.NewFlagSet("", flag.ExitOnError)
+	fl.Usage = func() {
+		fmt.Fprint(flag.CommandLine.Output(), `
+Generate a new private key.
+
+When successful a new file with binary content containing private key is
+created. This command fails if the private key file already exists.
+`)
+		fl.PrintDefaults()
+	}
+	var (
+		keyPathFl = fl.String("key", env("BNSCLI_PRIV_KEY", os.Getenv("HOME")+"/.bnsd.priv.key"),
```

Comment:
I like this default along with the match in sign.

1. use `-key=..`
2. use BNSCLI_PRIV_KEY
3. use a standard path $HOME/.bnsd.priv.key

So if I never set anything it "just works" (like ~/.ssh/id_rsa) :+1: